ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -585,23 +595,45 @@ public Hash getHash() {
    * @return the size in bytes of the encoded transaction.
    */
   @Override
-  public int getSize() {
-    if (size == -1) {
-      memoizeHashAndSize();
+  public int getSizeForAnnouncement() {
+    if (sizeForAnnouncement == -1) {
+      memoizeSizeForAnnouncement();
     }
-    return size;
+    return sizeForAnnouncement;
   }
 
-  private void memoizeHashAndSize() {
+  /**
+   * Returns the size in bytes of the encoded transaction for block inclusion.
+   *
+   * @return the size in bytes of the encoded transaction for block inclusion.
+   */
+  @Override
+  public int getSizeForBlockInclusion() {
+    if (sizeForBlockInclusion == -1) {
+      memoizeHashAndSizeForBlockInclusion();
+    }
+    return sizeForBlockInclusion;
+  }
+
+  private void memoizeHashAndSizeForBlockInclusion() {
     final Bytes bytes = TransactionEncoder.encodeOpaqueBytes(this, EncodingContext.BLOCK_BODY);
     hash = Hash.hash(bytes);
-    if (transactionType.supportsBlob() && getBlobsWithCommitments().isPresent()) {
-      final Bytes pooledBytes =
-          TransactionEncoder.encodeOpaqueBytes(this, EncodingContext.POOLED_TRANSACTION);
-      size = pooledBytes.size();
-      return;
+    sizeForBlockInclusion = bytes.size();
+    if (!transactionType.supportsBlob() || this.getBlobsWithCommitments().isEmpty()) {
+      sizeForAnnouncement = sizeForBlockInclusion;
+    }
+  }
+
+  private void memoizeSizeForAnnouncement() {
+    final Bytes pooledBytes =
+        TransactionEncoder.encodeOpaqueBytes(this, EncodingContext.POOLED_TRANSACTION);
+    sizeForAnnouncement = pooledBytes.size();
+    if (!transactionType.supportsBlob() || this.getBlobsWithCommitments().isEmpty()) {
+      sizeForBlockInclusion = pooledBytes.size();
+      if (hash == null) {
```

Comment:
can you add a comment on why this is being done

Reply:
Done